dask/dataframe/optimize.py

```diff
@@ -147,8 +147,10 @@ def _walk_deps(dependents, key, success):
             if key == success:
                 return True
             deps = dependents[key]
-            if deps:
-                return all(_walk_deps(dependents, dep, success) for dep in deps)
+            if deps:  # noqa: B023
+                return all(
+                    _walk_deps(dependents, dep, success) for dep in deps  # noqa: B023
+                )
```

Comment:
Do we need `# noqa: B023` twice here?

Reply:
For this solution, yes. Unfortunately, `B023` is raised separately on both the if statement and the for loop (below):
```bash
$ flake8 --select B023
dask/dataframe/optimize.py:150:16: B023 Function definition does not bind loop variable 'deps'.
dask/dataframe/optimize.py:152:44: B023 Function definition does not bind loop variable 'dep'.
dask/dataframe/optimize.py:152:69: B023 Function definition does not bind loop variable 'deps'.
```
There is probably a better solution, though I'm not sure what it is.

Reply:
Okay, no worries. Just wanted to make sure we couldn't easily remove one of them